guides/hack/16-readonly/01-introduction.md

```diff
@@ -39,8 +39,8 @@ function test(readonly Bar $x) : void {
 }
 ```
 
-### How is `readonly` different from contexts and capabilities that control property mutation (such as write_props)?
-[Contexts](https://docs.hhvm.com/hack/contexts-and-capabilities/available-contexts-and-capabilities) such as `write_props` affect an entire function (and all the functions it calls), whereas readonly affects specific values / expressions. 
+### How is `readonly` different from contexts and capabilities that control property mutation (such as `write_props`)?
+[Contexts](https://docs.hhvm.com/hack/contexts-and-capabilities/available-contexts-and-capabilities) such as `write_props` affect an entire function (and all of its callees), whereas readonly affects specific values / expressions. 
```

Comment:
can you remove `https://docs.hhvm.com/` from this please, so links are local (and show local changes) when developing